src/averageAges.js

```diff
@@ -55,10 +85,27 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const mothers = people.filter(person => person.sex === 'f');
+
+  const whoHasMothers = people.filter((person) => {
+    return onlyWithSon
+      ? (person.sex === 'm' && person.mother !== null)
+      : person.mother !== null;
+  });
+
+  const differenceAgeChildWithMother = [];
+
+  whoHasMothers.find((child) => {
```

Comment:
Are you sure you need the find method here? This method searches for and returns an element.